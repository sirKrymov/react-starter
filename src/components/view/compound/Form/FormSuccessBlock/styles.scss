@import '../../../../../styles/mixins';

.form-success-block {
  max-width: 360px;
  width: 100%;

  &__icon {
    height: 72px;
    width: 72px;
    @include center;
    border-radius: 50%;
    border: 3px solid var(--valid);
  }
}
