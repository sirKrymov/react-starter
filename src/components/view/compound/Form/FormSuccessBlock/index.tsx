// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import { MdDone } from 'react-icons/md';
import cn from 'classnames';

import { LinkButton } from 'components/view/atoms/LinkButton';
import { Typography } from 'components/view/atoms/Typography';
import { ReactIcon } from 'components/view/atoms/ReactIcon';
import { FormBlock } from '../FormBlock';

import './styles.scss';

interface IProps {
  backLink?: string;
  text: string;
}

export function FormSuccessBlock({
  backLink = '/',
  text
}: IProps): ReactElement {
  const formSuccessBlockClass = cn({
    'form-success-block': true
  });

  const successIconClass = cn({
    'form-success-block__icon': true
  });

  return (
    <div className={formSuccessBlockClass}>
      <FormBlock marginBottom="s6" alignH="center">
        <div className={successIconClass}>
          <ReactIcon color="valid" size="xxxl">
            <MdDone />
          </ReactIcon>
        </div>
      </FormBlock>

      <FormBlock marginBottom="s6">
        <Typography variant="body1" align="center">
          {text}
        </Typography>
      </FormBlock>

      <LinkButton height="lg" to={backLink} id="form-success-block-link">
        Back to sign in
      </LinkButton>
    </div>
  );
}
