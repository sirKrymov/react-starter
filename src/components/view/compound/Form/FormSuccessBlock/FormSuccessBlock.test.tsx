import React from 'react';

import { BrowserRouter } from 'react-router-dom';
import renderer, { ReactTestRenderer } from 'react-test-renderer';

import { FormSuccessBlock } from '../FormSuccessBlock';

describe('<FormSuccessBlock> compound component', () => {
  let component: ReactTestRenderer;

  const createComponent = () => {
    component = renderer.create(
      <BrowserRouter>
        <FormSuccessBlock text="Success test" />
      </BrowserRouter>
    );
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(component.toJSON()).toBeDefined();
    });

    it('THEN it should match snapshot', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
