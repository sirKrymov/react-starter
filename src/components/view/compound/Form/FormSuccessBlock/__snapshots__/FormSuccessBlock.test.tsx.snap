// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSuccessBlock> compound component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="form-success-block"
>
  <div
    className="form-block form-block--mb-s6 form-block--align-center"
  >
    <div
      className="form-success-block__icon"
    >
      <div
        className="react-icon react-icon--color-valid react-icon--size-xxxl"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="form-block form-block--mb-s6"
  >
    <p
      className="typography typography--display-initial typography--variant-body1 typography--color-textPrimary typography--align-center typography--font-roboto"
    >
      Success test
    </p>
  </div>
  <div
    className="ui-size ui-size--height-lg ui-size--width-full"
  >
    <a
      className="link-button button--font-weight-bold link-button--variant-contained link-button--color-secondary"
      href="/"
      id="form-success-block-link"
      onClick={[Function]}
    >
      <p
        className="typography typography--display-initial typography--variant-inherit typography--color-inherit typography--align-left typography--font-roboto"
      >
        Back to sign in
      </p>
    </a>
  </div>
</div>
`;
